### PR TITLE
Fix vsphere cluster ytt template  multi cluster issue

### PR DESCRIPTION
**Pull request by XudongLiuHarold**
Signed-off-by: Xudong Liu <xudongl@vmware.com>

### What this PR does / why we need it

This PR fixes when using AVI to provide control plane VIP, vsphere cluster ytt template multi overlay error.

The bug was introduced by this PR: https://github.com/vmware-tanzu/tanzu-framework/pull/1108

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note
NONE
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by saji-pivotal — APPROVED** [review]
Thanks for making the fix! Can you please update the Testing section in the PR description? LGTM otherwise

File: pkg/v1/providers/infrastructure-vsphere/v1.0.1/ytt/overlay.yaml [hunk 1]
```diff
@@ -109,7 +109,7 @@ kind: VSphereCluster
 metadata:
   name: #@ data.values.CLUSTER_NAME
 spec:
-  #@ if not data.values.AVI_CONTROL_PLANE_HA_PROVIDER:
+  #@ if data.values.AVI_CONTROL_PLANE_HA_PROVIDER != "true":
```
Comment (HanFa) [inline pkg/v1/providers/infrastructure-vsphere/v1.0.1/ytt/overlay.yaml:112]: can you change line 94 as well? `if (not data.values.AVI_CONTROL_PLANE_HA_PROVIDER)`

Comment (tschwaller) [reply pkg/v1/providers/infrastructure-vsphere/v1.0.1/ytt/overlay.yaml:112]: There are many other places in the file using 
`not data.values.AVI_CONTROL_PLANE_HA_PROVIDER`
Do they also need to be changed?

Comment (HanFa) [reply pkg/v1/providers/infrastructure-vsphere/v1.0.1/ytt/overlay.yaml:112]: Hi Tom, after testing, it seems that we should treat `AVI_CONTROL_PLANE_HA_PROVIDER` as boolean. So no need to change this line. (correct way is still `if not AVI_CONTROL_PLANE_HA_PROVIDER`)

**Comment by HanFa** [comment]
@XudongLiuHarold could you close this PR since its dup has been merged?

**Pull request closed without merging**